src/Language/Visitor.php

```diff
@@ -211,36 +212,28 @@ public static function visit($root, $visitor, $keyMap = null)
                 $parent = array_pop($ancestors);
 
                 if ($isEdited) {
-                    if ($inArray) {
-                        // $node = $node; // arrays are value types in PHP
-                        if ($node instanceof NodeList) {
-                            $node = clone $node;
-                        }
-                    } else {
+                    if (! is_array($node)) {
                         $node = clone $node;
                     }
                     $editOffset = 0;
-                    for ($ii = 0; $ii < count($edits); $ii++) {
-                        $editKey   = $edits[$ii][0];
-                        $editValue = $edits[$ii][1];
-
-                        if ($inArray) {
+                    if ($inArray) {
+                        foreach ($edits as $editKey => $editValue) {
                             $editKey -= $editOffset;
```

Comment:
I watched this carefully in the debugger, and it was never anything but `0`. Is it important?